### PR TITLE
Remove check that preds value need to be smaller than num_classes

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/metrics/issues/354
Remove that check that `preds` needs to be smaller than `num_classes` which does not make sense since we added support for logits. Updated tests case by scaling the `logit` test case with a factor 10 (to make sure generated values are bigger than `num_classes=5`).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Review by tchaton — APPROVED** [review]
LGTM !
